great_expectations/rule_based_profiler/types/serializable_builder.py

```diff
@@ -0,0 +1,46 @@
+import json
+from typing import Any
+
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.types import SerializableDictDot
+from great_expectations.util import deep_filter_properties_iterable
+
+
+class SerializableBuilder(SerializableDictDot):
+    """
+    A SerializableBuilder provides methods to serialize any builder object of a rule generically.
+    """
+
+    def to_dict(self) -> dict:
+        dict_obj: dict = {}
+
+        field_name: str
+        field_value: Any
+        for field_name, field_value in deep_filter_properties_iterable(
+            properties=super().to_dict(),
+        ).items():
+            # exclude non-serializable fields that a rule builder may contain
+            if (
+                field_name
+                not in [
+                    "_data_context",
+                ]
+                and field_name[0] == "_"
+                and hasattr(self, field_name)
+            ):
+                field_name = field_name[1:]
+                dict_obj[field_name] = field_value
```

Comment:
This logic feels super susceptible to breakage in my opinion - why not offset the burden of implementation to the child classes and specify the logic there?

I love generalizing and reusing code but I'm wary of catch all code here.

Let's also split up this logic into different lines to help with debugging (`deep_filer_properties_iterable`, `super().to_dict()`, `items()` iteration, all on separate lines)

Reply:
@cdkini Let us leave this one for discussion to agree on what makes sense to do here.  There are a few trade-offs.